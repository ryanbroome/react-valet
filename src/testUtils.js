import React from "react";
import UserContext from "./auth/UserContext";
// import { render } from "@testing-library/react";
// import { MemoryRouter } from "react-router";
import App from "../src/App";
// import TransactionList from "./TransactionList";

const MockUserContext = ({ children }) => {
  const mockToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InRlc3R1c2VyIiwibG9jYXRpb25JZCI6MSwiaXNBZG1pbiI6ZmFsc2UsImlhdCI6MTcxMzA3NTM4Mn0.dXlnawfzk6wlsgpouoSp58QLvlOOSJFC03iq_FXab-I";
  const mockUserDetail = {
    userId: 1,
    vehicleId: 1,
    locationId: 1,
    transactionId: 1,
    ticketNum: "1",
    mobile: "5553334444",
    color: "red",
    make: "honda",
    username: "testuser",
  };
  const mockLocation = {};
  const mockVehicle = {};
  const mockTransaction = {};

  return <UserContext.Provider value={{ token: mockToken, userDetail: mockUserDetail, location: mockLocation, vehicle: mockVehicle, transaction: mockTransaction, update: App.update }}>{children}</UserContext.Provider>;
};

export { MockUserContext };
