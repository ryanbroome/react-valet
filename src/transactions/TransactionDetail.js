import { Card, CardBody, CardTitle, CardSubtitle, ListGroup, ListGroupItem, CardText, Button } from "reactstrap";

import React, { useEffect, useState, useContext } from "react";
import { useParams, useHistory, Link } from "react-router-dom";
import ValetApi from "../api/Api";

const TransactionDetail = () => {
  const { id } = useParams();
  const [transaction, setTransaction] = useState();

  useEffect(
    function fetchTransactionsAtMount() {
      async function fetchTransactions() {
        try {
          const transactionRes = await ValetApi.getTransactionById(id);
          setTransaction(transactionRes.data.transactions);
        } catch (err) {
          console.log(err);
        }
      }
      fetchTransactions();
    },
    [id]
  );

  function viewTransaction(transaction) {
    return (
      <Card
        color="dark"
        body
        className="text-center">
        <CardBody>
          <CardSubtitle>{transaction.ticketNum}</CardSubtitle>
          <ListGroup flush>
            <ListGroupItem>Site: {transaction.sitename}</ListGroupItem>
            <ListGroupItem>Site ID: {transaction.locationId}</ListGroupItem>
            <ListGroupItem>Trans ID: {transaction.transactionId}</ListGroupItem>
            <ListGroupItem>Ticket:{transaction.ticketNum}</ListGroupItem>
            <ListGroupItem>Trans Time: {transaction.transactionTime}</ListGroupItem>
            <ListGroupItem>Mobile: {transaction.mobile}</ListGroupItem>
            <ListGroupItem>Vehicle ID: {transaction.vehicleId}</ListGroupItem>
            <ListGroupItem>Status: {transaction.vehicleStatus}</ListGroupItem>
            <ListGroupItem>In Time: {transaction.checkIn}</ListGroupItem>
            <ListGroupItem>Out Time: {transaction.checkOut}</ListGroupItem>
            <ListGroupItem>Color: {transaction.color}</ListGroupItem>
            <ListGroupItem>Make: {transaction.make}</ListGroupItem>
            <ListGroupItem>Damages: {transaction.damages}</ListGroupItem>
            <ListGroupItem>Notes: {transaction.notes}</ListGroupItem>
            <ListGroupItem>Valet ID: {transaction.userId}</ListGroupItem>
            <ListGroupItem>
              Valet: {transaction.firstName} {transaction.lastName}
            </ListGroupItem>
            <ListGroupItem>Last: {transaction.lastName}</ListGroupItem>
            <ListGroupItem>Phone: {transaction.phone}</ListGroupItem>
            <ListGroupItem>Email: {transaction.email}</ListGroupItem>
            <ListGroupItem>TotalParked: {Number(transaction.totalParked) > 0 ? transaction.totalParked : "None"}</ListGroupItem>
          </ListGroup>
        </CardBody>
      </Card>
    );
  }

  function showSpinner() {
    return (
      <div
        className="spinner-border text-primary"
        role="status">
        <span className="visually-hidden"></span>
      </div>
    );
  }
  return transaction ? viewTransaction(transaction) : showSpinner();
};

export default TransactionDetail;
