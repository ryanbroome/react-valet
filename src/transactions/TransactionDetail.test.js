import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import TransactionDetail from "../transactions/TransactionDetail";
import { MemoryRouter } from "react-router";
import { MockUserContext } from "../testUtils";

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <MockUserContext children={<TransactionDetail />}></MockUserContext>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("renders without crashing", function () {
  render(
    <MemoryRouter initialEntries={["/transactions/id/1"]}>
      <MockUserContext children={<TransactionDetail />}></MockUserContext>
    </MemoryRouter>
  );
});
