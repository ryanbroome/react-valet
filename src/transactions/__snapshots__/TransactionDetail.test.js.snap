// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionDetail: matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="spinner-border text-primary"
    role="status"
  >
    <span
      class="visually-hidden"
    />
  </div>
</DocumentFragment>
`;
